daemon/protocol/daemon.proto

```diff
@@ -27,31 +30,70 @@ message EchoReply {
   string pong = 1;
 }
 
-enum ProcessState {
-  UNKNOWN = 0;
-  PENDING = 1;
-  RUNNING = 2;
 
-  COMPLETED = 3;
-  FAILED = 4;
+// Create snapshot.
+message CreateSnapshotRequest {
+  string path = 1;
 }
 
-message Command {
-  repeated string argv = 1;
-  map<string, string> env = 2;
-  // Timeout in nanoseconds
-  int64 timeout = 3;
+message CreateSnapshotReply {
+  string error = 1;
 }
 
-message ProcessStatus {
-  string run_id = 1;
-  ProcessState state = 2;
-  string stdout_ref = 3;
-  string stderr_ref = 4;
-  int32 exit_code = 5;
-  string error = 6;
+
+// Checkout snapshot.
+message CheckoutSnapshotRequest {
+  string snapshot_id = 1;
+  string path = 2;
```

Comment:
FWIW, I could go either way on this... Maybe the path should just be part of the reply? WDYT?